### PR TITLE
Added model summary function

### PR DESCRIPTION
**Pull request by prince776**
I've added `summary()` function for `FFN` class. Please give suggestion if any changes are required. Based on those changes I'll also add this function to `RNN` and `BRNN` classes.

### CODE REVIEW

File: src/mlpack/methods/ann/ffn_impl.hpp [hunk 1]
```diff
@@ -496,6 +496,43 @@ void FFN<OutputLayerType, InitializationRuleType,
...
+  if (!deterministic)
+  {
+    deterministic = true;
+    ResetDeterministic();
+  }
+  Forward(std::move(inputs));
+
+  long long totalParams = 0;
+
+  std::cout << std::setfill('-') << std::setw(60) << '\n';
```
Comment (himanshupathak21061998) [inline src/mlpack/methods/ann/ffn_impl.hpp]: Instead of using `std::cout` use `Log::Info` .

Comment (prince776) [reply src/mlpack/methods/ann/ffn_impl.hpp]: Oh right. Thanks for pointing that out.

**Comment by prince776** [comment]
> Thanks for contribution @prince776 I am not sure about the usability of this function it is required or not member will give more insight about this.

Yes, I was also worried if this was necessary at all, but @khizirsiddiqui just said on gitter that it is a fairly useful feature. I also believe it's useful to understand and show to others how the input is getting transformed and what layers it is passing through and how heavy the NN is by getting the number of parameters to be trained. I especially use them in Kaggle notebooks and have seen many people use it.

**Comment by zoq** [comment]
I like the idea, however I was always wondering if we could get more if we would use model serialization as basis. That would provide weights, input/output/kernel size etc. as well. I think a more detailed summary would be nice to have, let me know what you think.

**Comment by prince776** [comment]
> I like the idea, however I was always wondering if we could get more if we would use model serialization as basis. That would provide weights, input/output/kernel size etc. as well. I think a more detailed summary would be nice to have, let me know what you think.

Yes, it'll be better to have a detailed summary. I just don't understand the concept of serialization that well. Can you give me some thing I could study, then I'll update the code.

**Comment by zoq** [comment]
We use boost for serialization: https://www.boost.org/doc/libs/1_72_0/libs/serialization/doc/index.html, the test cases should be helpful as well: https://github.com/mlpack/mlpack/blob/master/src/mlpack/tests/serialization_test.cpp.

**Comment by rcurtin** [comment]
This code touches the ANN code, and there was recently a big refactoring of the ANN code in #2259, so be sure to merge the master branch into your branch here to make sure that nothing will fail if this PR is merged. :+1: (I'm pasting this message into all possibly relevant PRs.)

**Pull request closed without merging**